GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -0,0 +1,375 @@
+                    {
+                        this.ExitCode = ReturnCode.GenericError;
+                    }
+                }
+                finally
+                {
+                    string cleanUpError = null;
+                    if (!this.TryRunCleanUp(out cleanUpError))
+                    {
+                        cleanUpError = Environment.NewLine + "WARNING: " + cleanUpError;
+                        this.output.WriteLine(cleanUpError);
+                        this.ExitCode = ReturnCode.Success;
+                    }
+                }
+            }
+            
+            if (this.ExitCode == ReturnCode.GenericError)
+            {
+                error = Environment.NewLine + "ERROR: " + error;
+                this.output.WriteLine(error);
+            }
+
+            this.output.WriteLine("Press Enter to exit.");
+            if (this.input == Console.In)
+            {
+                this.input.ReadLine();
+            }
+            
+            if (this.shouldExit)
+            {
+                Environment.Exit((int)this.ExitCode);
+            }
+        }
+
+        private bool LaunchInsideSpinner(Func<bool> method, string message)
+        {
+            return ConsoleHelper.ShowStatusWhileRunning(
+                method,
+                message,
+                this.output,
+                this.output == Console.Out && !GVFSPlatform.Instance.IsConsoleOutputRedirectedToFile(),
+                null);
+        }
+
+        private bool TryRunUpgradeInstall(out Version newVersion, out string error)
+        {
+            newVersion = null;
+
+            Version newGVFSVersion = null;
+            GitVersion newGitVersion = null;
+            string errorMessage = null;
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.TryCheckIfUpgradeAvailable(out newGVFSVersion, out errorMessage) ||
+                        !this.TryGetNewGitVersion(out newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.LogInstalledVersionInfo();
+                    this.LogVersionInfo(newGVFSVersion, newGitVersion, "Available Version");
+
+                    if (!this.preRunChecker.TryRunPreUpgradeChecks(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    if (!this.TryDownloadUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Downloading"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.preRunChecker.TryUnmountAllGVFSRepos(out errorMessage))

```

Comment:
Why is unmounting inside of a spinner called "Installing Git"?

Since you now have a separate spinner for remounting at the end, I'd be explicit about unmounting at the beginning as well. 

But of course now we run the risk of being overly verbose, so I'm open to suggestions on the best way to tune it... I just don't think we should misrepresent what we're doing, so we need to either change what's happening inside this spinner, or change its label.

Reply:
Have added "Unmounting repositories" spinner. To reduce the verbosity, I combined Installing Git and Installing GVFS into a single spinner. 

Here is the new messaging - 

```
Downloading...Succeeded
Unmounting repositories...Succeeded
Installing Git version: 2.17.1.gvfs.2.5, VFSForGit version: 1.0.18234.1...Succeeded
Remounting repositories...Succeeded
Press Enter to exit.
```